### ISSUE TITLE
Unable to have big prime field

### ISSUE THREAD

**Issue by Maldarowbi** [Macaulay2/M2#1476]
Here is an example of a prime number which is more than 64bit length.

kk = ZZ/18446744073709551616;
stdio:37:6:(3): error: ZZ/n not implemented yet for composite n

Would you please let me know if there is a possibility to work with a large field.

**Comment by DanGrayson** [comment]
Actually, your number is a power of 2, so here is an example that illustrates your problem:
```m2
i17 : isPrime (2^64+13)

o17 = true

i18 : ZZ/(2^64+13)
stdio:16:3:(3): error: expected argument 1 to be a small integer
```

Presumably you want to use such a field as coefficient ring for a polynomial ring.  @mikestillman can address whether there are any plans to implement such fields in the engine.

**Comment by mahrud** [comment]
@Maldarowbi you can use the [GF function](http://www2.macaulay2.com/Macaulay2/doc/Macaulay2-1.16/share/doc/Macaulay2/Macaulay2Doc/html/___G__F.html):
```m2
i1 : kk = GF(2, 64, Variable => a)

o1 = kk

o1 : GaloisField

i2 : R = kk[x,y,z]

o2 = R

o2 : PolynomialRing

i3 : (x+a)^64

      64    33    30    26    25    24    23    22    21    20    18    13    12
o3 = x   + a   + a   + a   + a   + a   + a   + a   + a   + a   + a   + a   + a  
     ---------------------------------------------------------------------------
        11    10    7    5    4    2
     + a   + a   + a  + a  + a  + a  + a + 1
```

**Comment by Maldarowbi** [comment]
Thanks @DanGrayson  ,

This is exactly what I get "stdio:16:3:(3): error: expected argument 1 to be a small integer"

I am working with a linear system Ax=b, where b is a 128-bit length. I want the field to be larger than the bitlength of b. A could be small numbers no problem.

Once I assigned the field I got this issue.

**Comment by Maldarowbi** [comment]
@mahrud 

I will try this one to see if it works.

**Comment by Maldarowbi** [comment]
Thanks @mahrud, 

Your suggestion works great works



i74 : kk = GF(2, 64, Variable => a);

i75 : n=100;

i76 : A= random(kk^n,kk^n) ** kk;

               100        100
o76 : Matrix kk    <--- kk

i77 : b= random(kk^n,kk^1) ** kk;

               100        1
o77 : Matrix kk    <--- kk

i78 : elapsedTime x=solve(A,b);
     -- 0.470183 seconds elapsed



Except for one thing, b is polinomials while I want them to be numbers. Please let me know if there is away to convert polynomial to a number and the otherway around.

**Comment by mahrud** [comment]
Well, they're polynomials in "a", which is a symbolic representative of the solution of some cyclotomic polynomial. In this case, I _think_ it would be this:
```
a^64 + a^33 + a^30 + a^26 + a^25 + a^24 + a^23 + a^22 + a^21 + a^20 + a^18 + a^13 + a^12 + a^11 + a^10 + a^7 + a^5 + a^4 + a^2 + a + 1
```
So I'm not sure what numerical representative you have in mind.

PS: I used sage to get the minimal polynomial of "a". I'm not sure how to get it in Macaulay2, but Givaro, the finite field arithmetic library used in M2, probably has a routine for it that we could expose.

**Comment by d-torrance** [comment]
This was fixed in #3218:

```m2
i1 : ZZ/18446744073709551616

             ZZ[]
o1 = --------------------
     18446744073709551616

o1 : QuotientRing

i2 : ZZ/(2^64+13)

             ZZ[]
o2 = --------------------
     18446744073709551629

o2 : QuotientRing
```
